app/src/main/java/com/alexP/assignment1/ui/myProfileActivity/MyProfileActivity.kt

```diff
@@ -29,49 +29,34 @@ class MyProfileActivity : BaseActivity<ActivityMyProfileBinding>() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        lifecycleScope.launch {
-            val email = readEmail(dataStore)
-            binding?.textViewNameSurname?.text = parseEmail(email!!)
-        }
+        viewModel =
+            ViewModelProvider(this, MyProfileViewModelFactory(this))[MyProfileViewModel::class.java]
 
-        binding?.let {
+        binding.let {
             Glide.with(this)
```

Comment:
You already have function ImageView.loadCircularImage(). Replace Glide snippet with your extension function.